tests/io/vasp/test_inputs.py

```diff
@@ -888,29 +888,44 @@ def test_static_constructors(self):
         kpoints = Kpoints.gamma_automatic((3, 3, 3), [0, 0, 0])
         assert kpoints.style == Kpoints.supported_modes.Gamma
         assert kpoints.kpts == [(3, 3, 3)]
+        assert all(isinstance(kpt, int) for kpt in kpoints.kpts[0])
+
         kpoints = Kpoints.monkhorst_automatic((2, 2, 2), [0, 0, 0])
         assert kpoints.style == Kpoints.supported_modes.Monkhorst
         assert kpoints.kpts == [(2, 2, 2)]
-        kpoints = Kpoints.automatic(100)
+        assert all(isinstance(kpt, int) for kpt in kpoints.kpts[0])
```

Comment:
could be refactored to `{*map(type, kpoints.kpts[0])} == {int}` but maybe less readable

Reply:
Yes it looks less readable, also I guess [type checking with `type(var) == type_of_var` feels a bit off](https://stackoverflow.com/a/152596/24021108)?

I don't think `kpt` would be a subclass of `int` ATM so it would be safe, but still not sure if we want to change this. I would let you decide :) Thanks for the comment.